packages/plugin-node-surrounding-code/surrounding-code.js

```diff
@@ -0,0 +1,43 @@
+const SURROUNDING_LINES = 3
+const { readFile } = require('fs')
+
+module.exports = {
+  init: client => {
+    const loadSurroundingCode = stackframe => new Promise((resolve, reject) => {
+      try {
+        if (!stackframe.lineNumber || !stackframe.file) return resolve(stackframe)
+        getSurroundingCode(stackframe.file, stackframe.lineNumber, (err, code) => {
+          if (err) return resolve(stackframe)
+          stackframe.code = code
+          return resolve(stackframe)
+        })
+      } catch (e) {
+        return resolve(stackframe)
+      }
+    })
+
+    client.config.beforeSend.push(report => new Promise((resolve, reject) => {
+      Promise.all(report.stacktrace.map(loadSurroundingCode))
+        .then(stacktrace => {
+          report.stacktrace = stacktrace
+          resolve()
+        })
+        .catch(reject)
+    }))
+  }
+}
+
+const getSurroundingCode = (file, lineNumber, cb) => {
+  readFile(file, 'utf8', (err, data) => {
```

Comment:
Will JS files always be encoded in UTF-8?

Reply:
In node, I believe this to be the case. This SO seems to back that belief up, but not conclusively: https://stackoverflow.com/questions/10125141/what-encoding-is-expected-for-node-js-source-code

In any case, the most likely other encoding is `ascii` which is a subset of `utf8`.

The other options that node support for reading files are `binary`, `hex`, `base64` and `utf16le` – but the first three would be valid source code so wouldn't run. The latter may result in working source code and could encounter encoding issues if we interpret as `utf8`. I don't know if anyone has ever encoded a node source file in `utf16le` 🤔

Reply:
As I imagine this is the most common case and the previous code assumed this, I don't think any further action is required on this